### PR TITLE
[WAL-481] add pin code screen before re-opening application

### PR DESCRIPTION
**Pull request by hrehman-sphereon**

### CODE REVIEW

File: src/store/actions/user.actions.ts [hunk 1]
```diff
@@ -70,3 +71,10 @@ export const setActiveUser = (userId: string): ThunkAction<Promise<void>, RootSt
       .catch(() => dispatch({type: SET_ACTIVE_USER_FAILED}));
   };
 };
+
+export const removeActiveUser = (): ThunkAction<Promise<void>, RootState, unknown, Action> => {
```
Comment (BtencateSphereon) [inline src/store/actions/user.actions.ts:75]: rename this to logout, and rename the setActiveUser to login. also do this for all the types associated with these actions.

File: src/store/actions/user.actions.ts [hunk 2]
```diff
@@ -70,3 +71,10 @@ export const setActiveUser = (userId: string): ThunkAction<Promise<void>, RootSt
...
+  return async (dispatch: ThunkDispatch<RootState, unknown, Action>) => {
+    dispatch({type: USERS_LOADING});
+    dispatch({type: REMOVE_ACTIVE_USER_SUCCESS, payload: null});
```
Comment (BtencateSphereon) [inline src/store/actions/user.actions.ts:78]: payload should be undefined, not null

File: src/handlers/LockingHandler/index.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import {removeActiveUser} from '../../store/actions/user.actions';
+
+class LockingHandler {
+  private lockingEventListener: NativeEventSubscription;
+
+  public enableLocking = async (): Promise<void> => {
+    const handleAppStateChange = async (nextAppState: string): Promise<void> => {
+      if (nextAppState === 'background' || nextAppState === 'active') {
+        debug('Locking application...');
+        console.log('Locking application...');
```
Comment (BtencateSphereon) [inline src/handlers/LockingHandler/index.ts:19]: remove the console log

**Review by BtencateSphereon — CHANGES_REQUESTED** [review]

File: src/store/actions/user.actions.ts (hunk 1 above)
Comment (hrehman-sphereon) [reply src/store/actions/user.actions.ts:75]: @nklomp @Brummos I guess ... making it logout is not right ... Let's say I logged out ... and now want to login with a different user. We have not designed the wallet to handle that... I suggest to keep it the same for now.

In the scope of this 3 story point ticket doing that kind of refactor will increase the testing surface too big ... even the VC and contacts may need to be checked ...

Comment (nklomp) [reply src/store/actions/user.actions.ts:75]: It is just a name suggestion, as the removeUser is highly confusing

Comment (hrehman-sphereon) [reply src/store/actions/user.actions.ts]: renamed

File: src/store/actions/user.actions.ts (hunk 2 above)
Comment (hrehman-sphereon) [reply src/store/actions/user.actions.ts]: changed

File: src/handlers/LockingHandler/index.ts (hunk 3 above)
Comment (hrehman-sphereon) [reply src/handlers/LockingHandler/index.ts]: removed

**Review by BtencateSphereon — APPROVED** [review]
LGTM

**Pull request merged**